### ISSUE TITLE
Feature Request: Sort tabs alphabetically

### ISSUE THREAD

**Issue by fgotzens** [spyder-ide/spyder#5911]
## Description of your problem

I see that it's been a long way to get the tabs in Spyder to become moveable: https://github.com/spyder-ide/spyder/issues/564 

However, it would be really useful if Spyder would either (a) provide a standard setting to sort them alphabetically or (b) at least save the position according to the users preferences.
When a project consists of 10 or more .py-files, at some point searching for the correct tabs becomes really unhandy...

## Versions and main components

* Spyder Version: 3.2.4
* Python Version: 2.7.14 64bits
* Qt Version: 5.6.2
* PyQt Version: 5.6
* Operating system: Windows 7 x64 Professional

**Comment by ccordoba12** [comment]
That's why we added a file switcher to the Editor, which can be activated with `Ctrl+P` by default. Is that not enough?

**Comment by fgotzens** [comment]
My intention is just to contribute ideas to an easier and more efficient coding environment. When my currently activated tab is in the middle of, say, 10 open tabs and I'm looking for a module which I cannot see directly in the tab row, I have to search for it, since I don't know whether it is located to the left or to the right from my current tab position.
Unfortunately, pressing `Ctrl+P` is not really helping me, because also there one needs to scroll (since not all modules fit in) and they are not sorted as well...

My current workaround is to manually (!) sort the modules in the `.spyproject/workspace.ini` file and I was just wondering if this could be achieved somehow automatically directly in Spyder?

**Comment by ccordoba12** [comment]
@fgotzens, it's not a bad idea, but we don't have time to implement it. You're welcome to send us a pull request to do it, though.

**Comment by luciansmith** [comment]
This is something I would like to see as well, in some form.  Perhaps the simplest would be to alphabetize the list of tabs that you get when you click the folder icon to the left of the tabs (the folder with the down-arrow on it).

I didn't know about control-P, but it seems to be the same list--if it could be alphabetized, that'd be great.

(To be clear, I don't want to move the tabs around, I just want the *list* of tabs to be sorted.  All other behavior would remain the same.)

**Comment by JakobJakobson** [comment]
How about reorganizing the function ```setup_file_list``` in ```spyder/spyder/widgets/fileswitcher.py``` in a way that the file name is the first argument? That way the results gets sorted by their name.

```python
    def setup_file_list(self, filter_text, current_path):
        """Setup list widget content for file list display."""
        short_paths = shorten_paths(self.paths, self.save_status)
        paths = self.paths
        icons = self.icons
        results = []
        trying_for_line_number = ':' in filter_text

        # Get optional line number
        if trying_for_line_number:
            filter_text, line_number = filter_text.split(':')
            if line_number == '':
                line_number = None
            # Get all the available filenames
            scores = get_search_scores('', self.filenames,
                                       template="<b>{0}</b>")
        else:
            line_number = None
            # Get all available filenames and get the scores for
            # "fuzzy" matching
            scores = get_search_scores(filter_text, self.filenames,
                                       template="<b>{0}</b>")

        # Get max width to determine if shortpaths should be used
        max_width = self.get_item_size(paths)[0]
        self.fix_size(paths)

        # Build the text that will appear on the list widget
        for index, score in enumerate(scores):
            text, rich_text, score_value = score
            if score_value != -1:
                text_item = '<big>' + rich_text.replace('&', '') + '</big>'
                if trying_for_line_number:
                    text_item += " [{0:} {1:}]".format(self.line_count[index],
                                                       _("lines"))
                if max_width > self.list.width():
                    text_item += u"<br><i>{0:}</i>".format(short_paths[index])
                else:
                    text_item += u"<br><i>{0:}</i>".format(paths[index])
                if (trying_for_line_number and self.line_count[index] != 0 or
                        not trying_for_line_number):
                    results.append((text_item, score_value, index))   # changing the order of these results can give you a ordered file list

        # Sort the obtained scores and populate the list widget
        self.filtered_path = []
        plugin = None
        for result in sorted(results):
            index = result[2] # has to be updated to reflect the changed order of the results
            path = paths[index]
            icon = icons[index]
            text = ''
            try:
                title = self.widgets[index][1].get_plugin_title().split(' - ')
                if plugin != title[0]:
                    plugin = title[0]
                    text += '<br><big><b>' + plugin + '</b></big><br>'
                    item = QListWidgetItem(text)
                    item.setToolTip(path)
                    item.setSizeHint(QSize(0, 25))
                    item.setFlags(Qt.ItemIsEditable)
                    self.list.addItem(item)
                    self.filtered_path.append(path)
            except:
                # The widget using the fileswitcher is not a plugin
                pass
            text = ''
            text += result[0] # has also be adapted to reflect the change in the  order of the results
            item = QListWidgetItem(icon, text)
            item.setToolTip(path)
            item.setSizeHint(QSize(0, 25))
            self.list.addItem(item)
            self.filtered_path.append(path)
        
        # To adjust the delegate layout for KDE themes
        self.list.files_list = True

        # Move selected item in list accordingly and update list size
        if current_path in self.filtered_path:
            self.set_current_row(self.filtered_path.index(current_path))
        elif self.filtered_path:
            self.set_current_row(0)

        # If a line number is searched look for it
        self.line_number = line_number
        self.goto_line(line_number)
```

**Comment by JakobJakobson** [comment]
Actually it can be done more elegant by using a lambda function in the ```sorted``` function.

```python
...
        # Sort the obtained scores and populate the list widget
        self.filtered_path = []
        plugin = None
        for result in sorted(results, key=lambda fbw: fwb[2]):
...
```

In this case you just have to adapt the sort function and can leaf the rest of the code untouched.

**Comment by JakobJakobson** [comment]
See  #6174 for a possible implementation.

**Comment by jnsebgosselin** [comment]
@ccordoba12 milestone?

**Comment by ccordoba12** [comment]
Done.

**Comment by jnsebgosselin** [comment]
@fgotzens @luciansmith If you could take a look at https://github.com/spyder-ide/spyder/issues/7993 and give your feedback please, that would be much appreciated. Otherwise, this issue may get closed as resolved after PR #8015 is merged.

**Comment by luciansmith** [comment]
Thanks for the heads-up!  It does indeed look like a good solution to me.

**Comment by ccordoba12** [comment]
This was fixed by PR #9109.
